File: Greedy/1449번 - 수리공 항승/1449.py
import sys

input = sys.stdin.readline

n, l = map(int, input().split())
pipe = list(map(int, input().split()))

def solve():
    pipe.sort()
    tape, tape_len = 0, 0
    
    for i in pipe:
        if tape_len < i:
            tape_len = i + l - 1
            tape += 1
    
    print(tape)

if __name__ == '__main__':
    solve()

File: Greedy/1449번 - 수리공 항승/1449.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n, l;
vector<int> pipe;

void input() {
    cin >> n >> l;
    for(int i = 0; i < n; i++) {
        int tmp;
        cin >> tmp;
        pipe.push_back(tmp);
    }
}

void solve() {
    sort(pipe.begin(), pipe.end());
    
    int tape = 0, len = 0;
    
    for(int i = 0; i < n; i++) {
        if(pipe[i] > len) {
            len = pipe[i] + l - 1;
            tape++;
        }
    }
    
    cout << tape << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}